smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/commit/KafkaRedisCheckpointCommit.java

```diff
@@ -0,0 +1,112 @@
+import io.vertx.mutiny.redis.client.Redis;
+import io.vertx.mutiny.redis.client.RedisAPI;
+import io.vertx.mutiny.redis.client.Response;
+import io.vertx.redis.client.RedisOptions;
+
+/**
+ * This implementation is probably wrong.
+ */
+public class KafkaRedisCheckpointCommit extends KafkaCheckpointCommit {
+
+    private final Redis redis;
+
+    private RedisAPI redisAPI;
+
+    public KafkaRedisCheckpointCommit(KafkaConnectorIncomingConfiguration config,
+            Vertx vertx,
+            KafkaConsumer<?, ?> consumer,
+            BiConsumer<Throwable, Boolean> reportFailure,
+            int defaultTimeout,
+            Redis redis) {
+        super(vertx, config, consumer, reportFailure, defaultTimeout);
+        this.redis = redis;
+    }
+
+    private <T> Uni<T> runWithRedis(Function<RedisAPI, Uni<T>> action) {
+        return Uni.createFrom().item(this.redisAPI)
+                .onItem().ifNull().switchTo(this.redis.connect().map(RedisAPI::api).onItem().invoke(api -> this.redisAPI = api))
+                .flatMap(action::apply);
+    }
+
+    @ApplicationScoped
+    @Identifier("checkpoint-redis")
+    public static class Factory implements KafkaCommitHandler.Factory {
+
+        @Override
+        public KafkaCommitHandler create(KafkaConnectorIncomingConfiguration config, Vertx vertx,
+                KafkaConsumer<?, ?> consumer, BiConsumer<Throwable, Boolean> reportFailure) {
+            int defaultTimeout = config.config()
+                    .getOptionalValue(ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG, Integer.class)
+                    .orElse(60000);
+            String redisConnectionString = config.config().getValue("redis-connection-string", String.class);
+            Redis redis = Redis.createClient(vertx, new RedisOptions()
+                    .setMaxPoolSize(30)
+                    .setMaxWaitingHandlers(1024)
+                    .addConnectionString(redisConnectionString));
+
+            return new KafkaRedisCheckpointCommit(config, vertx, consumer, reportFailure, defaultTimeout, redis);
+        }
+
+    }
+
+    @Override
+    public void terminate(boolean graceful) {
+        super.terminate(graceful);
+        if (redisAPI != null) {
+            redisAPI.close();
+        }
+        redis.close();
+    }
+
+    @Override
+    protected Uni<ProcessingState<?>> fetchProcessingState(TopicPartition partition) {
+        return runWithRedis(api -> api.get(getKey(partition))
+                .onFailure().invoke(t -> log.errorf(t, "Error fetching processing state %s", partition))
+                .onItem().invoke(r -> log.debugf("Fetched state for partition %s : %s", partition, r))
+                .map(r -> Optional.ofNullable(r)
+                        .map(Response::toBuffer)
+                        .map(this::deserializeState)
+                        .orElse(null)));
+    }
+
+    private String getKey(TopicPartition partition) {
+        return partition.topic() + "-" + partition.partition();
```

Comment:
Seems that Redis use ':' are default delimited. But `-` is fine (really a question of taste).

Reply:
👍